pkgs/development/python-modules/pytest/default.nix

```diff
@@ -3,8 +3,35 @@
 , atomicwrites, mock, writeText, pathlib2
 }:
 
-let generic = { version, sha256 }:
-  buildPythonPackage rec {
+let generic = { version, sha256 }: let
+
+  # runTests is a function that returns a string that can be used as a checkPhase.
+  runTests = with stdenv.lib; {
+    disabledTests ? [],     # Disable tests that match expression.
+    disabledTestIds ? [],   # Disable tests that match given node id's. This is useful in case a test in a specific file needs to be disabled.
+    options ? [],           # py.test options.
+    targets ? [],           # py.test files or folders.
+    variables ? {},         # Environment variables to export.
```

Comment:
`variables` is a bit ambiguities. How about `environment` or `environmentVariables`?